presto-main/src/main/java/com/facebook/presto/sql/planner/plan/TableFinishNode.java

```diff
@@ -67,7 +67,6 @@ public TableFinishNode(
     {
         super(sourceLocation, id, statsEquivalentPlanNode);
 
-        checkArgument(target != null || source instanceof TableWriterNode);
```

Comment:
I don't think you should be removing the second part of this assertion.

Reply:
This is a tricky one. Two cases:

1. If target is null, it throws a NullPointerException on line 72
2. If target is not null, the `||` means that source does not have to be an instance of TableWriterNode, so we don't want to make that check.

I.e. removing the first but not the second condition would be a behavior change that makes this method throw where it previously did not.

Reply:
It is possible the existing code is buggy. In particular, maybe it shouldn't be making the null check on line 72. What do you think?

Reply:
@pranjalssh can you help us understand what the intention behind this check was?